src/main/java/com/databricks/jdbc/api/impl/DatabricksResultSet.java

```diff
@@ -595,9 +605,27 @@ public boolean isFirst() throws SQLException {
     return executionResult.getCurrentRow() == 0;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>This method uses different strategies based on the result set type:
+   *
+   * <ul>
+   *   <li>For {@link LazyThriftResult} instances: Checks if there are no more rows available (using
+   *       {@code hasNext()}), since the total row count is unknown until all rows are fetched.
+   *   <li>For other result types: Compares the current row position against the known total row
+   *       count.
+   * </ul>
+   *
+   * @return {@code true} if the cursor is on the last row, {@code false} otherwise
+   * @throws SQLException if the result set is closed or an error occurs
+   */
   @Override
   public boolean isLast() throws SQLException {
     checkIfClosed();
+    if (executionResult instanceof LazyThriftResult) {
+      return !executionResult.hasNext();
```

Comment:
isLast should return true only when we are on last row, not when there are no more rows
```suggestion
      return executionResult.getCurrentRow() >= 0 && ! executionResult.hasNext();
```

Reply:
>true only when we are on last row

isn't  !executionResult.hasNext() doing the same thing? or are you referring to cases like empty result set, etc.?